src/components/QueryEditor/VisualQueryEditor/FilterItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { AdxColumnSchema, KustoQuery } from '../../../types';
+import {
+  QueryEditorExpressionType,
+  QueryEditorOperatorExpression,
+} from 'components/LegacyQueryEditor/editor/expressions';
+import {
+  getOperatorExpressionOptions,
+  getOperatorExpressionValue,
+  setOperatorExpressionName,
+  setOperatorExpressionProperty,
+  setOperatorExpressionValue,
+} from './utils/utils';
+import { isMulti, toOperatorOptions } from './utils/operators';
+import { columnsToDefinition, valueToDefinition } from 'schema/mapper';
+import { QueryEditorPropertyType } from 'schema/types';
+
+interface FilterItemProps {
+  datasource: AdxDataSource;
+  query: KustoQuery;
+  filter: Partial<QueryEditorOperatorExpression>;
+  columns: AdxColumnSchema[] | undefined;
+  templateVariableOptions: SelectableValue<string>;
+  onChange: (item: QueryEditorOperatorExpression) => void;
+  onDelete: () => void;
+}
+
+const FilterItem: React.FC<FilterItemProps> = (props) => {
+  const { datasource, query, filter, onChange, onDelete, columns, templateVariableOptions } = props;
+
+  const loadValues = async () => {
+    if (!filter.property?.name) {
+      return [];
+    }
+
+    return datasource
+      .autoCompleteQuery(
+        {
+          ...query,
+          search: {
+            operator: { name: 'isnotempty', value: '' },
+            property: { name: filter.property?.name, type: filter.property.type },
+            type: QueryEditorExpressionType.Operator,
+          },
+        },
+        columns
+      )
+      .then((result: string[]) => {
+        if (Array.isArray(templateVariableOptions.options) && filter.property?.type === 'string') {
+          // When selecting a string, automatically quote template variables to generate a valid syntax
+          templateVariableOptions.options = templateVariableOptions.options.map((op: SelectableValue<string>) => {
+            return { label: op.value, value: `'${op.value}'` };
+          });
+        }
+        return result.map((r): SelectableValue<string> => ({ label: r, value: r })).concat(templateVariableOptions);
+      });
+  };
+
+  const [state, loadOptions] = useAsyncFn(loadValues, [query, filter.property]);
+
+  return (
+    <InputGroup>
+      <Select
+        aria-label="column"
+        width="auto"
+        value={filter.property?.name ? valueToDefinition(filter.property?.name) : null}
+        options={columns ? columnsToDefinition(columns) : []}
+        allowCustomValue
+        onChange={(e) => e.value && onChange(setOperatorExpressionProperty(filter, e.value, e.type))}
+      />
+      <Select
+        aria-label="operator"
+        width="auto"
+        value={filter.operator?.name && toOption(filter.operator.name)}
+        options={toOperatorOptions(filter.property?.type)}
+        onChange={({ value }) => value && onChange(setOperatorExpressionName(filter, value))}
+      />
+      <div hidden={filter.property?.type !== QueryEditorPropertyType.Number}>
+        <AutoSizeInput
```

Comment:
This logic shows a numeric input if the property is a number. The normal thing here would be to do:

```ts
{filter.property?.type === QueryEditorPropertyType.Number} && <AutoSizeInput
```

But I found out that the component `EditorList` freaks out if children appear / disappear, causing them to be wrongly resized. The only solution I found that worked was to wrap these component in `div`s and hide them.